### PR TITLE
[UMD] Fixed T6832

### PR DESCRIPTION
**Pull request by AurelioDeRosa**

### CODE REVIEW

File: packages/babel-plugin-transform-es2015-modules-umd/src/index.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ export default function ({ types: t }) {
           let moduleName = args.length === 3 ? args.shift() : null;
           let amdArgs = call.arguments[0];
           let func = call.arguments[1];
+          let browserGlobals = arguments[1].opts && arguments[1].opts.browserGlobals || {};
```
Comment (thejameskyle) [inline packages/babel-plugin-transform-es2015-modules-umd/src/index.js]: Why are you referencing `arguments[1]`?

File: packages/babel-plugin-transform-es2015-modules-umd/src/index.js [hunk 2]
```diff
@@ -62,9 +63,8 @@ export default function ({ types: t }) {
             } else if (arg.value === "exports") {
               return t.memberExpression(t.identifier("mod"), t.identifier("exports"));
             } else {
-              return t.memberExpression(t.identifier("global"), t.identifier(
-                t.toIdentifier(basename(arg.value, extname(arg.value)))
-              ));
+              let identifier = _path.basename(arg.value, _path.extname(arg.value));
+              return t.memberExpression(t.identifier("global"), browserGlobals[identifier] ? t.identifier(t.toIdentifier(browserGlobals[identifier])) : t.identifier(t.toIdentifier(identifier)));
```
Comment (thejameskyle) [inline packages/babel-plugin-transform-es2015-modules-umd/src/index.js]: Can you split this up over multiple lines with actual if statements to make it more readable?

File: packages/babel-plugin-transform-es2015-modules-umd/src/index.js (hunk 1 above)
Comment (AurelioDeRosa) [inline packages/babel-plugin-transform-es2015-modules-umd/src/index.js]: What should I use instead?

File: packages/babel-plugin-transform-es2015-modules-umd/src/index.js (hunk 2 above)
Comment (AurelioDeRosa) [inline packages/babel-plugin-transform-es2015-modules-umd/src/index.js]: Yes, I'll update the PR.

File: packages/babel-plugin-transform-es2015-modules-umd/src/index.js (hunk 1 above)
Comment (thejameskyle) [inline packages/babel-plugin-transform-es2015-modules-umd/src/index.js]: Just add another param to the function?

Comment (AurelioDeRosa) [inline packages/babel-plugin-transform-es2015-modules-umd/src/index.js]: I thought you meant some already defined variable I wasn't aware of.

On a side note, there's no need to provide this kind of answers. From the experience (jQuery team), if you start a reply with "Just do [PUT SENTENCE HERE]", not only it isn't useful for the contributor but you also sound a lot like a... you can guess what. While I'm able to understand what you mean (I'm definitely not a beginner), this kind of answers are very discouraging and embarrassing for beginners. They feel they are missing something obvious and there are high chances they will not contribute again.

**Comment by hzoo** [comment]
This looks fine other than that we should add a test case for this. Also since we are adding a plugin option we should update the readme/docs on the website.

**Comment by AurelioDeRosa** [comment]
It seems that @clayreimann has done the work already with PR #3366. You can favor that PR instead of this.

**Comment by hzoo** [comment]
Thanks @AurelioDeRosa

**Pull request closed without merging**